atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -175,6 +175,31 @@ Iterator<Map.Entry<Cell, byte[]>> getSortedColumns(
     @Idempotent
     Map<Cell, byte[]> get(TableReference tableRef, Set<Cell> cells);
 
+    /**
+     * Similar to {@link #get(TableReference, Set)}, but allows the caller to specify the number of  expected present
+     * cells. This is important to allow clients that might have schemas that guarantees only a subset of the columns
+     * are present to still benefit from being able to skip the immutable timestamp lock check on non empty
+     * reads of thoroughly swept tables.
+     *
+     * If we find values for the exact number of expected present cells, we'll skip the immutable timestamp lock check.
+     * If we find less values, we'll perform the immutable timestamp lock check.
+     * If we find more value, we'll throw an exception, as it means that either:
```

Comment:
nit: `values` (also double space above) - bunch of typos here, fine for a draft though we should fix before merge